### PR TITLE
Do not assume the source dir is the working dir

### LINKED ISSUE #359: Graph CLI to accept argument to path of subgraph.yaml
We use the same subgraph repo for a number of similar subgraphs. Yet the CLI is limited in that it expects a file `./subgraph.yaml` in the root of the repo. In order to support our workflow, we'd like the path to the `yaml` file to be optionally provided to `graph-cli` as an argument. This applies to the `build`, `codegen` and `deploy` actions. 

This is how we're currently working around the limitation. https://github.com/Synthetixio/synthetix-subgraph/blob/master/package.json#L10-L13

### PR DESCRIPTION
**Pull request by leoyvens**
We were doing that in some places. Resolves #359.

### CODE REVIEW

File: src/compiler.js [hunk 1]
```diff
@@ -52,6 +52,7 @@ class Compiler {
         return true
       }
     } catch (e) {
+      console.log(e)
```
Comment (Jannis) [inline src/compiler.js]: Do we want this log message? _If_ we log anything here, it should be done using the `print` module of the Gluegun toolbox.

**Review by Jannis — CHANGES_REQUESTED** [review]
One minor remark about `console.log`, everything else looks good to me. I confirmed the bug on master.

File: src/compiler.js (hunk 1 above)
Comment (leoyvens) [reply src/compiler.js]: I think usually this should never happen, I hit it due to a double variable declaration while developing. Changed this to use gluegun.

**Review by Jannis — APPROVED** [review]

File: src/compiler.js (hunk 1 above)
Comment (Jannis) [reply src/compiler.js]: Ah, yeah, any errors during compilation should already be logged inside the `try {}` branch.

**Pull request merged**